src/output-flex.cpp

```diff
@@ -494,12 +494,13 @@ void output_flex_t::write_row(table_connection_t *table_connection,
             if (geom.empty()) {
                 write_null(copy_mgr, column);
             } else {
+                // if srid of the area column is the same as for the geom column
                 double const area =
-                    get_options()->reproject_area
-                        ? ewkb::parser_t(geom).get_area<reprojection>(
-                              get_options()->projection.get())
-                        : ewkb::parser_t(geom)
-                              .get_area<osmium::geom::IdentityProjection>();
+                    column.srid() == srid
+                        ? ewkb::parser_t(geom)
+                              .get_area<osmium::geom::IdentityProjection>()
+                        : ewkb::parser_t(geom).get_area<reprojection>(
+                              reprojection::create_projection(srid).get());
```

Comment:
This looks rather expensive. You create (as in allocate) a reprojection object every time a row is written.

Reply:
Yes, this is expensive. But this code is hopefully soon going to be revised anyway when we clean up the geometry generation with a cache holding already created geometries. I opted to leave this in for the time being. In most cases where this is used the projection for the `area` column will be the same as for the geometry column anyway, in which case the first option with the hard-coded `IdentityProjection` is used. So the expensive path here will only be taken if you use a non-mercator projection and use an area column.

Reply:
I disagree that geometry generation will automatically fix that. The problem at hand here is that this PR switches from having one global projection object to instantiating them locally with potentially different projections. This is orthogonal to having multiple geometries (which would still all use the same global projection if it wasn't for this PR). Fixing this requires to reconsider how the projection code is called, possibly getting rid of the shared object and having something lightweight that can be instantiated on the fly.